### DISCUSSION TITLE
Channel parameter for caiman.load?

### DISCUSSION THREAD

**Discussion by ethanbb** [flatironinstitute/CaImAn#1366] [General]
I'm just curious why the channel parameter for `load` was disabled:

```python
if channel is not None:
        logging.error('movies.py:load(): channel parameter is not supported for single movie input')
```

It seems very useful for multi-channel recordings and is used in `load_movie_chain`. However, to get it to work in `load_movie_chain`, I would have to return a movie from `load` with channels rather than frames in the first dimension, which seems like it would break more things (currently trying to `load` a multi-channel sbx recording will just throw an error). The `channel` could just be ignored within `load` for any file formats that don't support it, no?

**Comment by pgunn** [comment]
The channel code is ancient and never was documented (probably never saw non-internal use). There's a lot that could be done to make it more useful (including documenting it, and implementing it for more formats), or we could just remove it.

Either would be fine by me (I'm not likely to extend and document it myself though).

**Reply by ethanbb** [reply]
OK, I see how it might make sense to remove it. When I first read the code, I thought this was a full-fledged part of the API (and wrote the sbx-loading code with this in mind). It's a shame we don't have more history from when it was written, because I'm curious to know what barriers came up to getting it to work with other formats, if any.

Expecting any multi-channel/obscure formats to be converted to single-channel and standard formats before loading into CaImAn is probably reasonable. My only hesitation is that for sbx files in particular, this is now the only barrier to being able to load them directly (for multi-channel recordings; single-channel ones already work). So I wonder if making some minimal changes to get that to work would be worth it, even if we don't try to make it work for every format.
